software/contrib/bernoulli_gates.py

```diff
@@ -0,0 +1,180 @@
+from europi import *
+from time import sleep_ms
+from random import random
+from europi_script import EuroPiScript
+import machine
+
+# Constant values for display
+BAR_LEN = 50
+BAR_WID = 6
+INDI_WID = 5
+
+# Trigger Time (some module may not be able to catch up the trigger if it is too short)
+TRG_T = 20
+
+# Overclock the Pico for improved performance.
+machine.freq(250_000_000)
+
+class SingleBernoulliGate():
+    def __init__(self, id):
+        self.mode_flg = 0 # mode 0: Trigger, mode 1: Gate, mode 2: Toggle
+        self.coin = 0 # probability
+        self.right_possibility = 0 # change every iteration
+        self.left_possibility = 0
+        self.right_possibility_sampled = 0 # only change when triggered
+        self.left_possibility_sampled = 0
+        self.id = id
+        if self.id == 0:
```

Comment:
This `id` parameter is fine for now, but if we ever wanted to add a third gate I think we would want to make all of these fields constructor params. (For example if an expander module added the right ins and outs to support a third gate.) The idea here would be to separate the responsibilities of 'how a gate works' from 'which i/o it uses and where on the screen it displays'. 

If we get the bernoulli gate class isolated enough it could be possible to pull it up into a library class where other scripts could build upon it. I'm not saying that you should be thinking about future uses like this when writing scripts, I'm just throwing ideas out that people might be interested in running with.

Reply:
Thanks for the advice, please see my modification.